invenio_rdm_records/services/config.py

```diff
@@ -54,10 +55,11 @@
     value_labels=VocabularyLabels('languages')
 )
 
-subject_facet = TermsFacet(
-    field='metadata.subjects.id',
+subject_facet = NestedTermsFacet(
```

Comment:
See comment in inveniosoftware/invenio-app-rdm#1036. The rest of the PR makes full sense independent of if we choose to go with TermsFacet or NestedTermsFacet.

Reply:
Is it ok if I make a ticket for that one and we can revisit? Issues I see:

- We have a lot of subjects at NU (at least 2 other categories than MeSH will be coming) and displaying them all as 1 is awkward (huge list (or with caching not big enough!) and searchers can't drill down/ focus on category of interest) 
- Resource types would have subfaceting but not this one; it's a little strange 
- Heavier: have to rely on `read_all` + more development time